File: src/pocketcloud/event/impl/player/PlayerEvent.php
<?php

namespace pocketcloud\event\impl\player;

use pocketcloud\event\Event;
use pocketcloud\player\CloudPlayer;

abstract class PlayerEvent extends Event {

    public function __construct(private readonly CloudPlayer $player) {}

    public function getPlayer(): CloudPlayer {
        return $this->player;
    }
}

File: cloud-shared/src/main/java/de/pocketcloud/shared/event/player/PlayerEvent.java
package de.pocketcloud.shared.event.player;

import de.pocketcloud.api.component.player.ICloudPlayer;
import de.pocketcloud.api.event.Event;

public abstract class PlayerEvent implements Event {

    protected final ICloudPlayer player;

    public PlayerEvent(ICloudPlayer player) {
        this.player = player;
    }
}